security/src/main/java/io/micronaut/security/authentication/DefaultAuthorizationExceptionHandler.java

```diff
@@ -45,23 +47,33 @@
 
     private final RedirectConfiguration redirectConfiguration;
     private final PriorToLoginPersistence priorToLoginPersistence;
+    private final ErrorResponseProcessor<?> responseProcessor;
 
     /**
      * Default constructor.
      */
     public DefaultAuthorizationExceptionHandler() {
-        this.redirectConfiguration = null;
-        this.priorToLoginPersistence = null;
+        this(null, null, null);
     }
 
     /**
      * @param redirectConfiguration Redirect configuration
      * @param priorToLoginPersistence Persistence mechanism to redirect to prior login url
      */
-    @Inject
     public DefaultAuthorizationExceptionHandler(RedirectConfiguration redirectConfiguration, @Nullable PriorToLoginPersistence priorToLoginPersistence) {
+        this(redirectConfiguration, priorToLoginPersistence, null);
```

Comment:
Instead of `null` can you manually instantiate `HateosErrorResponseProcessor` which is the default implementation of `ErrorResponseProcessor`?